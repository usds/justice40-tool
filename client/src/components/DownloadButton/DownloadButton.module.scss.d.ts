declare namespace DownloadButtonNamespace {
    export interface IDownloadButtonScss {
      downloadButtonLink: string;
      buttonComponent: string;
      buttonComponentYellow: string;
      buttonComponentGray: string;
      buttonContainer: string;
      buttonText: string;
      buttonImageBlue: string;
      buttonImageYellow: string;
      buttonImageGray: string;
    }
  }

declare const DownloadButtonScssModule: DownloadButtonNamespace.IDownloadButtonScss & {
    /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
    locals: DownloadButtonNamespace.IDownloadButtonScss;
  };

  export = DownloadButtonScssModule;
