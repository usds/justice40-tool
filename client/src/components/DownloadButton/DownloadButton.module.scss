@use '../../styles/design-system.scss' as *;

@mixin buttonImageBase {
    width: 21px;
    margin-top: -3px;
};

@mixin downloadButtonBase {
    height: 40px;
    @include u-margin-top(3);
    
    .buttonContainer {
        display: flex;

        .buttonText {
            @include u-margin-right(1);
            }
        
        .buttonImageYellow {
            @include buttonImageBase();
            filter: invert(13%) sepia(76%) saturate(5142%) hue-rotate(192deg) brightness(80%) contrast(106%);
        }

        .buttonImageGray {
            @include buttonImageBase();
            filter: invert(100%) sepia(0%) saturate(7489%) hue-rotate(232deg) brightness(101%) contrast(101%);
        }

        .buttonImageBlue {
            @include buttonImageBase();
            filter: invert(100%) sepia(100%) saturate(1%) hue-rotate(137deg) brightness(103%) contrast(101%);
        }
    }
}

.downloadButtonLink{
    display: flex;
    text-decoration: none;
    justify-content: center;

    .buttonComponentYellow {
        @include downloadButtonBase;
        @include u-text("blue-70v");
        @include u-bg("yellow-20v");
        &:hover {
            @include u-bg("yellow-20");
            @include u-text("gray-90");
            }
    }

    .buttonComponentGray {
        @include downloadButtonBase;
        @include u-text("white");
        @include u-bg("gray-60");
        &:hover {
            @include u-bg("gray-90");
            @include u-text("gray-10");
            }
    }
    
    .buttonComponent {
       @include downloadButtonBase;
    }
}
