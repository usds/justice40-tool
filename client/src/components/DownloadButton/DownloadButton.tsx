import React from 'react';
import {Button} from '@trussworks/react-uswds';

import * as styles from './DownloadButton.module.scss';
// @ts-ignore
import fileDownloadIcon from '/node_modules/uswds/dist/img/usa-icons/file_download.svg';

export interface IDownloadButtonProps {
  downloadLink: string,
  buttonText: string
  imageAltTagText: string,
  color : 'gray' | 'yellow' | 'default'
}

const DownloadButton = ({downloadLink, buttonText, imageAltTagText, color}: IDownloadButtonProps) => {
  return (
    <a className={styles.downloadButtonLink} href={downloadLink} download>
      <Button
        type="button"
        className={
          color === 'yellow' ? styles.buttonComponentYellow :
          color === 'gray' ? styles.buttonComponentGray :
          styles.buttonComponent
        }>
        <div className={styles.buttonContainer}>
          <div className={styles.buttonText}>
            {buttonText}
          </div>
          <img
            className={
              color === 'yellow' ? styles.buttonImageYellow :
              color === 'gray' ? styles.buttonImageGray :
              styles.buttonImageBlue}
            src={fileDownloadIcon}
            alt={imageAltTagText}
          />
        </div>
      </Button>
    </a>
  );
};

export default DownloadButton;
