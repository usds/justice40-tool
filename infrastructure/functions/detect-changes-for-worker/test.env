REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1