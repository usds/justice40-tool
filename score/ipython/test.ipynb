{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
