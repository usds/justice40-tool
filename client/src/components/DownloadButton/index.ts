import DownloadButton from './DownloadButton';
export default DownloadButton;
