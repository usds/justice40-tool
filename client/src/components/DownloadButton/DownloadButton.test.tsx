import React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import DownloadButton from './DownloadButton';

describe('rendering of DownloadButton Component', () => {
  const {asFragment} = render(
      <LocalizedComponent>
        <DownloadButton
          downloadLink='https://google.com'
          buttonText='Hello'
          imageAltTagText='download button'
        />
      </LocalizedComponent>,
  );
  it('checks if component renders', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
