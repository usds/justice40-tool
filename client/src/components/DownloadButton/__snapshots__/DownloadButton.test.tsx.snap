// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of DownloadButton Component checks if component renders 1`] = `
<DocumentFragment>
  <a
    download=""
    href="https://google.com"
  >
    <button
      class="usa-button"
      data-testid="button"
      type="button"
    >
      <div>
        <div>
          Hello
        </div>
        <img
          alt="download button"
          src="test-file-stub"
        />
      </div>
    </button>
  </a>
</DocumentFragment>
`;
